xmppserver/src/main/java/org/jivesoftware/util/SystemProperty.java

```diff
@@ -231,6 +231,8 @@ public void xmlPropertyDeleted(final String property, final Map<String, Object>
     private final boolean sorted;
 
     private SystemProperty(final Builder<T> builder) {
+        // Before we do anything, convert XML based provider setup to Database based
+        JiveGlobals.migrateProperty(builder.key);
```

Comment:
This prevents SystemProperty to be used without incident for elements that need to remain in the XML, I think. I'm not saying that's bad perse, just wondering if that's intended.

Reply:
Yes, intended. I can’t see any reason for properties to remain in XML and have DB values too. Many places were calling the migrateProperty method, it made sense to me to do so in the one place. 

But yes, should have made that a bit more explicit.

Reply:
My only concern is that there might be properties that _need_ to remain in the XML (database connection strings, fqdn), that once being handled as a `SystemProperty`, get unexpectedly migrated.

Reply:
I struggle to see a circumstance where a property would need to be in both XML and database. Which value is correct? Placing DB connection strings in the DB is of little use (in fact I recently made a change to stop that happening, as IME too many people were editing the DB values instead of the XML - https://github.com/igniterealtime/Openfire/blob/master/xmppserver/src/main/java/org/jivesoftware/openfire/XMPPServer.java#L165-L184), placing the FQDN in the database is wrong.

Where might a property be in both?

Reply:
I can't think of a property that should belong in both. I was worried that properties belonging in XML would accidentally be migrated to the database, by using the SystemProperty constructor.

Reply:
The SystemProperty class only supports database properties, so not sure how/why that could happen ...

Reply:
My lack of thoroughly understanding/reading the code confused this conversation.